File: client_leger/app/src/main/java/com/client_leger/colorimage/drawing/data/DrawEllipseData.kt
package com.client_leger.colorimage.drawing.data

data class DrawEllipseData(
    val drawPreview: Boolean,
    val strokeColor: String,
    val fillColor: String,
    val size: Int,
    val withStroke: Boolean,
    val withFill: Boolean,
    val isEven: Boolean,
    val beginning: Vec2,
    val end: Vec2,
    val text: String?,
    val textColor: String?
) : CommandData()

File: server/src/drawings/dto/commands/data/draw-ellipse-data.ts

import { IsBoolean, IsNumber, IsOptional, IsString } from 'class-validator';
import { Vec2 } from '../vec2';

export class DrawEllipseData {
    @IsBoolean()
    drawPreview: boolean;
    @IsString()
    strokeColor: string;
    @IsString()
    fillColor: string;
    @IsNumber()
    size: number;
    @IsBoolean()
    withStroke: boolean;
    @IsBoolean()
    withFill: boolean;
    @IsBoolean()
    isEven: boolean;
    beginning: Vec2;
    end: Vec2;

    @IsOptional()
    @IsString()
    textColor: string;
    @IsOptional()
    @IsString()
    text: string;
}
